doc/theory.md

```diff
@@ -52,3 +52,18 @@ Finally, FLOSS diffs the emulator memory states from before and
   6. Emulate decoder functions using extracted arguments and emulator state
   7. Diff memory state from before and after decoder emulation
   8. Extract human-readable strings from memory state difference
+
+## Go String Extraction
+The Go string extraction algorithm implemented in the FLOSS tool enables efficient retrieval of Go strings from binaries, simplifying reverse engineering and malware analysis tasks. By analyzing the struct string instances and leveraging length-sorted order, this algorithm accurately identifies the string blob. The surrounding byte examination and the use of cross-references further refine the extraction process. With this powerful algorithm, FLOSS empowers users to extract and analyze Go strings with ease, providing valuable insights into the inner workings of Go programs.
```

Comment:
i'd recommend a bit of background on why this is necessary. assume the reader doesn't know why special handling is necessary - can you give them a few sentences to explain what they'd otherwise see in the output?